patches/net/minecraft/network/codec/ByteBufCodecs.java.patch

```diff
@@ -13,3 +13,38 @@
              }
  
              public R decode(RegistryFriendlyByteBuf p_331253_) {
+@@ -519,9 +_,21 @@
+         return new StreamCodec<RegistryFriendlyByteBuf, HolderSet<T>>() {
+             private static final int NAMED_SET = -1;
+             private final StreamCodec<RegistryFriendlyByteBuf, Holder<T>> holderCodec = ByteBufCodecs.holderRegistry(p_332137_);
+-
++            private final Map<net.neoforged.neoforge.registries.holdersets.HolderSetType, StreamCodec<RegistryFriendlyByteBuf, ? extends net.neoforged.neoforge.registries.holdersets.ICustomHolderSet<T>>> holderSetCodecs = new java.util.HashMap<>();
```

Comment:
This needs to be a `ConcurrentHashMap` or an otherwise thread-safe map as singleplayer spawns two network threads which could both access this simultaneously.